dask/tests/test_base.py

```diff
@@ -129,6 +129,15 @@ def test_tokenize_numpy_scalar():
     ) == tokenize(np.array([(1, 2)], dtype=[("a", "i4"), ("b", "i8")])[0])
 
 
+@pytest.mark.skipif("not np")
+def test_tokenize_numpy_scalar_string_rep():
+    # Test tokenizing numpy scalars doesn't depend on their string representation
+    np.set_string_function(lambda x: "foo")
+    assert tokenize(np.array(1)) != tokenize(np.array(2))
```

Comment:
nitpick: a failure here would mean the `np.set_string_function(None)` isn't run, which may affect other tests.

I'd recommend putting the assert in a try / finally

```python
try:
    assert ...
finally:
    np.set_string_function(None)
```

or write a small fixture that does

```python
np.set_string_function(lambda x: "foo")
yield
np.set_string_function(None)
```

(pytest will make sure the block after the yield runs, even if the test fails / errors).

Reply:
Good catch, thanks for the suggestion! Just pushed a commit which adds `try`/`finally` accordingly